src/VisualStudio/CSharp/Test/Interactive/Commands/InteractiveCommandHandlerTests.cs

```diff
@@ -0,0 +1,122 @@
+@"int x;
+int y;");
+
+            // Tests cases when interactive buffer was not empty before calling the command.
+            // Execute in interactive clears the existing current buffer before execution.
+            AssertExecuteInInteractive(
+                @"{|Selection:var y = 2;$$|}",
+                "var y = 2;",
+                submissionBuffer: "var x = 1;");
+        }
+
+        [WpfFact]
+        [Trait(Traits.Feature, Traits.Features.Interactive)]
+        public void TestCopyToInteractive()
+        {
+            // Tests that the command is unavailable without a selection.
+            AssertUnavailableCopyToInteractive("$$");
+            AssertUnavailableCopyToInteractive($"{ExampleCode1}$$");
+            AssertUnavailableCopyToInteractive($"{ExampleCode1}$$", submissionBuffer: "var x = 1;");
+            AssertUnavailableCopyToInteractive($"{ExampleCode1}$$", submissionBuffer: "x = 2;");
+
+            // Tests a regular copy command.
+            AssertCopyToInteractive($"{{|Selection:{ExampleCode1}$$|}}", ExampleCode1);
+
+            // Tests cases when interactive buffer was not empty before calling the command.
+            AssertCopyToInteractive(
+                $"{{|Selection:{ExampleCode1}$$|}}",
+                $"var x = 1;\r\n{ExampleCode1}",
+                submissionBuffer: "var x = 1;");
+        }
+
+        private static void AssertUnavailableCopyToInteractive(string code, string submissionBuffer = null)
+        {
+            using (var workspace = InteractiveWindowCommandHandlerTestState.CreateTestState(code))
+            {
+                PrepareSubmissionBuffer(submissionBuffer, workspace);
+                Assert.Equal(CommandState.Unavailable, workspace.GetStateForCopyToInteractive());
+            }
+        }
+
+        private static void AssertCopyToInteractive(string code, string expectedBufferText, string submissionBuffer = null)
+        {
+            using (var workspace = InteractiveWindowCommandHandlerTestState.CreateTestState(code))
+            {
+                PrepareSubmissionBuffer(submissionBuffer, workspace);
+                workspace.SendCopyToInteractive();
+                Assert.Equal(expectedBufferText, workspace.WindowCurrentLanguageBuffer.CurrentSnapshot.GetText());
+            }
+        }
+
+        private static void AssertUnavailableExecuteInInteractive(string code)
+        {
+            using (var workspace = InteractiveWindowCommandHandlerTestState.CreateTestState(code))
+            {
+                Assert.Equal(CommandState.Unavailable, workspace.GetStateForExecuteInInteractive());
+            }
+        }
+
+        private static void AssertExecuteInInteractive(string code, string expectedSubmission, string submissionBuffer = null)
+        {
+            List<string> submissions = new List<string>();
+            EventHandler<string> appendSubmission = (_, item) => { submissions.Add(item.TrimEnd()); };
```

Comment:
This code is specific to tests to deal with trailing newlines executed by interactive window. For instance `IInteractiveWindow.SubmitAsync("code")` triggers `IInteractiveEvaluator.ExecuteCodeAsync("code\r\n")`.

Such trimming is also present in InteractiveWindowTests: [http://index/#Roslyn.InteractiveWindow.UnitTests/InteractiveWindowTests.cs,1042](http://index/#Roslyn.InteractiveWindow.UnitTests/InteractiveWindowTests.cs,1042).